### ISSUE TITLE
Documentation revamp

### ISSUE THREAD

**Issue by mxstbr** [keystonejs/keystone#3296]
TL;DR: We want to completely revamp the documentation, we're going to use [Gatsby](https://github.com/gatsbyjs/gatsby) for it and we need your help with guide writing!

We're planning to completely revamp the docs for the v4 release. @borisbozic is currently working on a complete redesign and we'll also change the way the documentation works and is structured.

These are the improvements we've come up with so far:
- **Move documentation back into main repo**: We moved the documentation out of the main repo into `keystonejs-site` in an effort to keep this one lean and small. This backfired, as now the docs are out of date and not synced at all. Instead, we want to put the site in a different repo but generate the docs from Markdown files in this one. @kylematthews is going to help us set this up with Gatsby.js.
- **Split documentation between API docs, Guides and technical docs**: Currently the documentation on the site is awesome if you're working _on_, but horrible if you're working _with_ Keystone. We need to split between user-facing docs (full-length guides, e.g. getting started, setting up a blog, setting up a portfolio site,… and API docs, i.e. how to use a certain field type) and development-facing docs (how the codebase is structured, how to set up a development version locally,…). A huge first start was made by @ostruzina and @dariasamo who migrated all the field type docs to this repo in #3261 and #3292, but this obviously isn't finished yet.

The main point where you can help is with guides. We want to have easy-to-follow guides for all common use cases and we need you to help us write them. **If you have a usecase and would be willing to write a short guide explaining how to set Keystone up for that please let us know in a comment!**

If you have any more suggestions on how we could improve the documentation please also let us know!

**Comment by dariasamo** [comment]
We can write a user-guide on Setting up a blog and cover:
- Running keystone locally
- Building a blog based on free Bootstrap template
a) Editing/adding jade templates
b) Creating new Models (for example, for pages like Contact, About)
c) Editing/adding routes
- Deploying result to Heroku

**Comment by christroutner** [comment]
I can help with:

   - Using the Admin API
   - Creating custom API interfaces
   - File and Image transfer via the Admin API

**Comment by Puine** [comment]
I made a small guidelines for myself to deploy a keystone project on an ubuntu server (with nginx), if it can be useful ? (from my point, the deploying part was the struggling part, and not well documented). But it's not directly keystone related ... not sure if it fits on the doc. 

Anyway, looking forward to test the new version !

**Comment by mxstbr** [comment]
Amazing, I've added the suggestions to the list above with your names next to them – we're currently working on figuring out how we're going to make this happen re: build process, will get back to you as soon as I know what format these docs should have!

> not sure if it fits on the doc.

Absolutely does, deployment is an integral part of every site!

**Comment by mxstbr** [comment]
Just finished talking to Kyle, all looking awesome. I've created a `docs/` folder, with a `guides/` folder beneath it – start filling that up!

**How to add a guide**

1. Create a new markdown file in `docs/guides/` called `yourtopic.md`
2. Fill it with your awesome guide!
3. Submit a Pull Request

Looking forward to the first guides dropping!

**Comment by christroutner** [comment]
So these guides are supposed to use markdown? It would be great if someone
could create a template for us to start with. If they want to create
standardized section headers, that would help make the guides more readable
too.

**Comment by mxstbr** [comment]
All of this will be done by the actual rendering, you don't need to pay attention to anything in the markdown except to use proper headings and [fenced code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/#fenced-code-blocks) :+1:

```Md
# My Guide

This is some content of my guide about keystone. Here is some information

## Setup

You need to create a markdown file.
```

**Comment by ivarni** [comment]
It's not much, but I'd like to see better docs for how to set up the keystone `nav` object because the current docs [confuse](https://github.com/keystonejs/keystone/issues/3287#issuecomment-238361293) me. I'd be happy to document what's missing from [the current](http://keystonejs.com/docs/configuration/#options-project) docs about admin navigation options. 

I'm not sure wether belongs in a *guide* or if it should just be part of the API docs though? If the latter is the case, where do I put it?

**Comment by christroutner** [comment]
You know would also be great? If someone could write up a guide on how to setup HTTPS as well setting up KeystoneJS for submitting encrypted input forms.

**Comment by mxstbr** [comment]
@wmertens may just have something for you @christroutner: https://medium.com/keystone-js/lets-encrypt-keystonejs-81fd55db9a3d#.k7q5o191y

**Comment by tomheathershub** [comment]
Sorry to polute the comments but  @Puine Any chance you could share your documentation with me, i'm looking to deploy keystone on an ubuntu server (with or without nginx)

**Comment by mxstbr** [comment]
The first guide for sending email is done! 🎉

Check it out: https://github.com/keystonejs/keystone/blob/master/docs/guides/how-to-send-emails.md

**Comment by JonForest** [comment]
Setting up Keystone over the weekend, I felt the docs light on detail.  Specifically, the views differ from Express view declarations, so some further explanation is required.
If you set up a basic Blog using Yeoman, the route/blog.js has three `view.on('init', function (next) {` event handlers declared, but there's nothing in the docs for:
 - state when this fires
 -  in what order these will execute
 - if there are any other events that can be listened to
 - which of these are 'per request' or 'per controller'

Just a more complete API reference is required I think, rather than just a couple of samples.

**Comment by KyleAMathews** [comment]
Hey folks, would love to hear what you think about the new site I'm working on! Check out the PR over at https://github.com/keystonejs/keystone/pull/3978

**Comment by Noviny** [comment]
To keep track #4102 fixes this now. Thanks a lot to @KyleAMathews for setting up gatsby for us and making this possible.

**Comment by Noviny** [comment]
Closing this as we have a new documentation site (and a new issue more closely tied to it)
